include/deal.II/base/quadrature_point_data.h

```diff
@@ -328,7 +328,7 @@ namespace parallel
     {
     public:
       DEAL_II_STATIC_ASSERT(std::is_base_of<TransferableQuadraturePointData, DataType>::value,
-                            "User's DataType class should be derived from QPData");
+                            "User's DataType class should be derived from QPData")
```

Comment:
The `do...while` construct can't appear in a place where a declaration may be -- say inside a class declaration :-(